src/shared-components/button/index.tsx

```diff
@@ -10,66 +10,74 @@ import LinkButton from './components/linkButton';
 import RoundButton from './components/roundButton';
 import TextButton from './components/textButton';
 import { COLORS, COLORS_PROP_TYPES } from '../../constants';
+import {
+  deprecatedProperties,
+  isLoadingPropFunction,
+} from './deprecatedPropsHanlder';
 
-const deprecatedProperties = {
-  loading: "The 'loading' prop is deprecated. Use 'isLoading' instead.",
-};
+export type ButtonType = 'primary' | 'secondary' | 'tertiary' | 'quaternary';
+// This is needed for RoundButton
+export type ButtonTypeWithAction = ButtonType | 'action';
```

Comment:
Shouldn't we define this in the `RoundButton` component? Or were there some complications with that.

Reply:
fair comment, I left it here because it is used in main styles and also Loader component, also thought leaving these 2 side by side would be more clear?

Reply:
That makes sense. I might rephrase the comment as:

```
/**
 * RoundButton can also accept a buttonType prop of 'action'
 */
```

But I'm also fine if you just resolve this thread without any changes!

Reply:
updated the comment, thanks for wording